editor/scene/3d/node_3d_editor_plugin.cpp

```diff
@@ -1573,6 +1573,35 @@ Transform3D Node3DEditorViewport::_compute_transform(TransformMode p_mode, const
 	}
 }
 
+void Node3DEditorViewport::_reset_transform(TransformType p_type) {
```

Comment:
You should add a check whether the selection is not empty, and return (without creating action) if it is.